cmd/package_managers.go

```diff
@@ -138,3 +202,115 @@ func fetchGitRepositoryFromRubyGems(packageName string, manager packageManagerCl
+		return "", err
+	}
+	nugetRegistrationBaseURL, err := getFieldFromIndexResults(nugetIndexResults.Resources,
+		"RegistrationsBaseUrl/3.4.0")
+	if err != nil {
+		return "", err
+	}
+
+	lastPackageVersion, err := getLatestListedVersion(nugetRegistrationBaseURL,
+		packageName, manager)
+	if err != nil {
+		return "", err
+	}
+
+	respPackageSpec, err := manager.Get(
+		nugetPackageBaseURL+"%[1]v/"+lastPackageVersion+"/%[1]v.nuspec", packageName)
+	if err != nil {
+		return "", sce.WithMessage(sce.ErrScorecardInternal,
+			fmt.Sprintf("failed to get nuget package spec json: %v", err))
+	}
+	defer respPackageSpec.Body.Close()
+
+	packageSpecResults := &nugetNuspec{}
+	err = xml.NewDecoder(respPackageSpec.Body).Decode(packageSpecResults)
+
+	if err != nil {
+		return "", sce.WithMessage(sce.ErrScorecardInternal,
+			fmt.Sprintf("failed to parse nuget nuspec xml: %v", err))
+	}
+	if packageSpecResults.Metadata == (nuspecMetadata{}) ||
+		packageSpecResults.Metadata.Repository == (nuspecRepository{}) ||
+		packageSpecResults.Metadata.Repository.URL == "" {
+		return "", sce.WithMessage(sce.ErrScorecardInternal,
+			fmt.Sprintf("source repo is not defined for nuget package %v", packageName))
+	}
+	return packageSpecResults.Metadata.Repository.URL, nil
+}
+
+func getLatestListedVersion(baseURL, packageName string, manager packageManagerClient) (string, error) {
+	resPackageRegistrationIndex, err := manager.Get(baseURL+"%s/index.json", packageName)
+	if err != nil {
+		return "", sce.WithMessage(sce.ErrScorecardInternal,
+			fmt.Sprintf("failed to get nuget package registration index json: %v", err))
+	}
+	defer resPackageRegistrationIndex.Body.Close()
+	packageRegistrationCatalogRoot := &nugetPackageRegistrationCatalogRoot{}
+	err = json.NewDecoder(resPackageRegistrationIndex.Body).Decode(packageRegistrationCatalogRoot)
+	if err != nil {
+		return "", sce.WithMessage(sce.ErrScorecardInternal,
+			fmt.Sprintf("failed to parse package registration index json: %v", err))
+	}
+	return getLatestListedVersionFromPackageRegistrationPages(packageRegistrationCatalogRoot.Pages, manager)
+}
+
+func getLatestListedVersionFromPackageRegistrationPages(pages []nugetPackageRegistrationCatalogPage,
+	manager packageManagerClient,
+) (string, error) {
+	for pageIndex := len(pages) - 1; pageIndex >= 0; pageIndex-- {
+		page := pages[pageIndex]
+		if page.Packages == nil {
+			respPage, err := manager.GetURI(page.ID)
+			if err != nil {
+				return "", sce.WithMessage(sce.ErrScorecardInternal,
+					fmt.Sprintf("failed to get nuget package registration page: %v", err))
+			}
+			defer respPage.Body.Close()
+			err = json.NewDecoder(respPage.Body).Decode(&page)
+
+			if err != nil {
+				return "", sce.WithMessage(sce.ErrScorecardInternal,
+					fmt.Sprintf("failed to parse nuget package registration page: %v", err))
+			}
+		}
+		for packageIndex := len(page.Packages) - 1; packageIndex >= 0; packageIndex-- {
+			if page.Packages[packageIndex].Entry.Listed {
```

Comment:
This may be a prerelease version. Is this desired? Here's some examples:
https://api.nuget.org/v3/registration5-gz-semver2/basetestpackage.searchfilters/index.json - latest version is prerelease

Also, there is something in SemVer 2.0.0 that is called metadata. It's the part after a `+` sign in the version which is used for informational purposes and not comparison purposes. Example:
https://api.nuget.org/v3/registration5-gz-semver2/serilog.exceptions/index.json
The latest version is `8.4.0+build.694`
The `+build.694` part is ignored for comparison purposes (per SemVer 2.0.0 spec) and is therefore not included when building URLs.

Does Go have a SemVer parsing library? You may able to get away with these tricks:
- First: If there is a `+` in the version string, remove it and everything after it. This takes a "full version string" in NuGet parlance to become a "normalized version version string".
- Second: If there is a `-` in the version string, it is a prerelease string.
- Third: If you are building a URL with this version, lowercase the string first. Thankfully a naive lowercase method will be fine here since package versions must only contain ASCII and all language I've seen have consistent lowercase foldering for ASCII characters.